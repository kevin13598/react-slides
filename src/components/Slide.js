import React from 'react';

export const Slide = ({
    slide, 
    increment, 
    decrement, 
    reset, 
    disablePrev, 
    disableNext
}) => {

    return (
        <div>
            <div id="navigation" className="text-center">
                <button data-testid="button-restart" className="small outlined" onClick={reset}>Restart</button>
                <button data-testid="button-prev" className="small" onClick={decrement} disabled={disablePrev}>Prev</button>
                <button data-testid="button-next" className="small" onClick={increment} disabled={disableNext}>Next</button>
            </div>
            <div id="slide" className="card text-center">
                <h1 data-testid="title">{slide.title}</h1>
                <p data-testid="text">{slide.text}</p>    
            </div>
        </div>
    )
}
