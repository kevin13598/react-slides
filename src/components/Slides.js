import React from 'react';
import { Slide } from './Slide';
import { slides } from '../data/slides';
import { useCounter } from './hooks/useCounter';

function Slides() {

    const {counter, increment, decrement, reset} = useCounter(0);

    return (
       <Slide
            slide = {slides[counter]}
            increment={increment}
            decrement = {decrement}
            reset = {reset}
            disablePrev = {(slides[counter] === slides[0])}
            disableNext = {(slides[counter] === slides[slides.length - 1])}
        />
    );

}

export default Slides;
