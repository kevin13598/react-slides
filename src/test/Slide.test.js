import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { Slide } from '../components/Slide';
import { renderHook } from '@testing-library/react-hooks';

describe('Pruebas en <Slide/>', () => {

    
    const slide = {
        title: "Today's workout plan",
        text: "We're gonna do 3 fundamental exercises."
    }
    
    const wrapper = shallow(<Slide slide={slide}/>);

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
});