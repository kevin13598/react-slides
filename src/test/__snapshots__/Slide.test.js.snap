// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Slide/> debe de mostrarse correctamente 1`] = `
<div>
  <div
    className="text-center"
    id="navigation"
  >
    <button
      className="small outlined"
      data-testid="button-restart"
    >
      Restart
    </button>
    <button
      className="small"
      data-testid="button-prev"
    >
      Prev
    </button>
    <button
      className="small"
      data-testid="button-next"
    >
      Next
    </button>
  </div>
  <div
    className="card text-center"
    id="slide"
  >
    <h1
      data-testid="title"
    >
      Today's workout plan
    </h1>
    <p
      data-testid="text"
    >
      We're gonna do 3 fundamental exercises.
    </p>
  </div>
</div>
`;
